docs/appendices/data-types.txt

```diff
@@ -0,0 +1,176 @@
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#TINYINT
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#numeric-types
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#SMALLINT
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#numeric-types
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#INTEGER
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#numeric-types
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#BIGINT
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#numeric-types
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#REAL
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#numeric-types
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#DOUBLE
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#string
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#VARCHAR
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#ip
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#VARCHAR
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#timestamp
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#TIMESTAMP
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#geo-point
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#ARRAY
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#geo-shape
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#JAVA_OBJECT
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#object
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#JAVA_OBJECT
+__ https://crate.io/docs/crate/reference/en/latest/general/ddl/data-types.html#array
+__ https://docs.oracle.com/javase/8/docs/api/java/sql/JDBCType.html#ARRAY
+
+Array Types
+-----------
+
+Array types map to `java.sql.Array`_ instances.
+
+Use the ``.getArray()`` to get the underlying array (it is usually safe to
+cast it to ``Object[]``) if you prefer. Otherwise, to have JDBC handle type
+mapping, you can use ``.getResultSet()`` and use the related
+``ResultSetMetaData`` to get the array values converted to Java POJOs.
+
+The ``ResultSet`` will have one column with the inner array type and the name
+of the array field (``Array.getBaseType()``) and as many rows as there are
+elements in the array.
+
+Here's one example:
+
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself) - please also add
+empty lines where it makes most sense
```

Comment:
please review this FIXME

Reply:
Ok, so the second part of the code below is incorrect so needs to be removed. 

The code block should be:

```
Statement statement = connection.createStatement();
ResultSet resultSet = statement.executeQuery(
    "SELECT array_field FROM my_table"
);
resultSet.first();
Array arrayField = resultSet.getArray("array_field");
Object[] arrayFieldValue = arrayFieldValue.getArray();
```

Reply:
As a personal preference, I'd remove "my_table" and replace it with a real life example (selecting tags from a blob post table or something similar, whatever would match our existing documentation pattern)